include/wally_crypto.h

```diff
@@ -482,6 +496,114 @@ WALLY_CORE_API int wally_ec_sig_to_public_key(
     unsigned char *bytes_out,
     size_t len);
 
+/**
+ * Verify that a secp256k1 scalar value is valid.
+ *
+ * :param scalar: The starting scalar to have a value added to.
+ * :param scalar_len: The length of ``scalar`` in bytes. Must be ``EC_SCALAR_LEN``.
+ */
+WALLY_CORE_API int wally_ec_scalar_verify(
+    const unsigned char *scalar,
+    size_t scalar_len);
+
+/**
+ * Add one secp256k1 scalar to another.
+ *
+ * :param scalar: The starting scalar to have a value added to.
+ * :param scalar_len: The length of ``scalar`` in bytes. Must be ``EC_SCALAR_LEN``.
+ * :param operand: The scalar value to add to ``scalar``.
+ * :param operand_len: The length of ``operand`` in bytes. Must be ``EC_SCALAR_LEN``.
+ * :param bytes_out: Destination for the resulting scalar.
+ * :param len: The length of ``bytes_out`` in bytes. Must be ``EC_SCALAR_LEN``.
+ *
+ * .. note:: Computes (scalar + operand) % G. Returns ``WALLY_ERROR`` if
+ *|    either input is not zero or within the secp256k1 group order G.
```

Comment:
Why do we fail if scalar is 0?

secp256k1 `scalar`s (sometimes referred to as `tweak`s) unlike secret keys can be 0. I think we should allow them to be 0.

Reply:
Zero is allowed, text is not (zero or within group order). I'll reword, cheers.